core/src/main/java/io/opencensus/tags/TagContext.java

```diff
@@ -47,30 +49,39 @@
   }
 
   /**
-   * Returns a builder based on this {@code TagContext}.
+   * Returns the current {@code TagContext}.
    *
-   * @return a builder based on this {@code TagContext}.
+   * @return the current {@code TagContext}.
    */
-  public Builder toBuilder() {
-    return new Builder(getTags());
+  public static TagContext getCurrentTags() {
+    return CurrentTagsUtils.getCurrentTagContext();
   }
 
   /**
-   * Returns an empty {@code TagContext}.
+   * Returns a new empty {@code Builder}.
    *
-   * @return an empty {@code TagContext}.
+   * @return a new empty {@code Builder}.
    */
-  public static TagContext empty() {
-    return EMPTY;
+  public static Builder emptyBuilder() {
```

Comment:
I'll try that design.  What about `TagsComponent` or `TaggingComponent` for the abstract class?  We may also need to add a factory for tag keys, if we need to enforce uniqueness of names.  Then TagsComponent would contain the TagContext and TagKey factories.